### ISSUE TITLE
Composer require FOSHttpCacheBundle fails even on fresh composer.json file.

### ISSUE THREAD

**Issue by piotrkochan** [FriendsOfSymfony/FOSHttpCacheBundle#504]
I try to install this package via composer and lost a lot of time, I even tried to use fresh crated `composer.json` and the error is still:

```composer require friendsofsymfony/http-cache-bundle guzzlehttp/psr7 php-http/guzzle6-adapter 
Using version ^2.6 for friendsofsymfony/http-cache-bundle
Using version ^1.4 for guzzlehttp/psr7
Using version ^2.0 for php-http/guzzle6-adapter
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for friendsofsymfony/http-cache-bundle ^2.6 -> satisfiable by friendsofsymfony/http-cache-bundle[2.6.0].
    - friendsofsymfony/http-cache-bundle 2.6.0 requires friendsofsymfony/http-cache ^2.5.2 -> satisfiable by friendsofsymfony/http-cache[2.5.2, 2.5.3, 2.5.4].
    - friendsofsymfony/http-cache 2.5.2 requires php-http/client-common ^1.1.0 -> satisfiable by php-http/client-common[1.3.0, 1.5.0, 1.7.0, 1.8.0, 1.8.1, v1.1.0, v1.2.0, v1.2.1, v1.4.0, v1.4.1, v1.4.2, v1.6.0].
    - friendsofsymfony/http-cache 2.5.3 requires php-http/client-common ^1.1.0 -> satisfiable by php-http/client-common[1.3.0, 1.5.0, 1.7.0, 1.8.0, 1.8.1, v1.1.0, v1.2.0, v1.2.1, v1.4.0, v1.4.1, v1.4.2, v1.6.0].
    - friendsofsymfony/http-cache 2.5.4 requires php-http/client-common ^1.1.0 -> satisfiable by php-http/client-common[1.3.0, 1.5.0, 1.7.0, 1.8.0, 1.8.1, v1.1.0, v1.2.0, v1.2.1, v1.4.0, v1.4.1, v1.4.2, v1.6.0].
    - php-http/client-common v1.1.0 requires php-http/httplug ^1.0 -> satisfiable by php-http/httplug[v1.0.0, v1.1.0].
    - php-http/client-common v1.2.0 requires php-http/httplug ^1.0 -> satisfiable by php-http/httplug[v1.0.0, v1.1.0].
    - php-http/client-common v1.2.1 requires php-http/httplug ^1.0 -> satisfiable by php-http/httplug[v1.0.0, v1.1.0].
    - php-http/client-common 1.3.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common 1.5.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common 1.7.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common 1.8.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common 1.8.1 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common v1.4.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common v1.4.1 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common v1.4.2 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - php-http/client-common v1.6.0 requires php-http/httplug ^1.1 -> satisfiable by php-http/httplug[v1.1.0].
    - Can only install one of: php-http/httplug[v2.0.0, v1.0.0].
    - Can only install one of: php-http/httplug[v2.0.0, v1.1.0].
    - php-http/guzzle6-adapter v2.0.0 requires php-http/httplug ^2.0 -> satisfiable by php-http/httplug[v2.0.0].
    - Installation request for php-http/guzzle6-adapter ^2.0 -> satisfiable by php-http/guzzle6-adapter[v2.0.0].
```

Probably it will work as described in https://github.com/FriendsOfSymfony/FOSHttpCacheBundle/issues/77 with @dev tag, but why it does not work as described in the documentation? https://foshttpcachebundle.readthedocs.io/en/latest/overview.html#installation

**Comment by piotrkochan** [comment]
This is bad, `@dev` does not install latest master version. See my composer.lock: https://pastebin.com/Ykipyccf

PHP: 
```
php -v                  
PHP 7.2.12-1+ubuntu18.04.1+deb.sury.org+1 (cli) (built: Nov 12 2018 09:55:44) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.2.0, Copyright (c) 1998-2018 Zend Technologies
    with Zend OPcache v7.2.12-1+ubuntu18.04.1+deb.sury.org+1, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v2.6.1, Copyright (c) 2002-2018, by Derick Rethans
```

**Comment by dbu** [comment]
the problem is with `Using version ^2.0 for php-http/guzzle6-adapter` - client-common has not been released as 2.0 yet, and 1.x requires httplug 1, while guzzle6-adapter 2 is already on httplug 2.

i hope there will be a release of client-common that supports httplug 2 soon - meanwhile, you can `composer require php-http/guzzle6-adapter:^1.1` which should work.
